sdk/nodejs/resource.ts

```diff
@@ -303,9 +339,13 @@ export abstract class ProviderResource extends CustomResource {
      * @param props The configuration to use for this provider.
      * @param opts A bag of options that control this provider's behavior.
      */
-    constructor(pkg: string, name: string, props?: Inputs, opts: ResourceOptions = {}) {
+    constructor(private readonly pkg: string, name: string, props?: Inputs, opts: ResourceOptions = {}) {
```

Comment:
I think unless we mark the property as `/*internal*/`, it will still cause issues with TypeScript when there is SxS - right?

(or do we no longer have to worry about that due to the blocks we put in place)?

Either way - maybe better to mark this internal?

Relatedly - should we guard code that uses `.getPackage` with a test for whether the method is there - providing a good error message if it is not?